keyvi/bin/keyvi_c/c_api.cpp

```diff
@@ -54,14 +54,14 @@ struct keyvi_dictionary {
 };
 
 struct keyvi_match {
-  explicit keyvi_match(const Match& obj) : obj_(obj) {}
+  explicit keyvi_match(const match_t& obj) : obj_(obj) {}
 
-  Match obj_;
+  match_t obj_;
 };
 
 struct keyvi_match_iterator {
-  explicit keyvi_match_iterator(const MatchIterator::MatchIteratorPair& obj) : current_(obj.begin()), end_(obj.end()) {}
-
+  explicit keyvi_match_iterator(const MatchIterator::MatchIteratorPair&& obj)
```

Comment:
```suggestion
  explicit keyvi_match_iterator(MatchIterator::MatchIteratorPair&& obj)
```

If the second `&` is added to move the iter, then we should drop the `const`.

Reply:
it doesn't move, but calls `begin()` and `end()` on it, so this should just be a normal ref.